Source/WebCore/dom/FullscreenManager.cpp

```diff
@@ -760,14 +758,12 @@ void FullscreenManager::emptyEventQueue()
     m_fullscreenErrorEventTargetQueue.clear();
 }
 
-void FullscreenManager::addDocumentToFullscreenChangeEventQueue(Document& document)
+void FullscreenManager::queueFullscreenChangeEventForDocument(Document& document)
 {
-    Node* target = document.fullscreenManager().fullscreenElement();
-    if (!target)
-        target = document.fullscreenManager().currentFullscreenElement();
-    if (!target)
-        target = &document;
-    m_fullscreenChangeEventTargetQueue.append(GCReachableRef(*target));
+    RefPtr target = document.fullscreenManager().fullscreenElement();
+    ASSERT(target);
+    document.fullscreenManager().addElementToChangeEventQueue(*target);
+    document.scheduleRenderingUpdate(RenderingUpdateStep::Fullscreen);
```

Comment:
Lets not leave a future null pointer dereference in our code, even if there's an ASSERT.